OpenRA.Mods.Common/UpdateRules/Rules/20231010/ReplaceCloakPalette.cs

```diff
@@ -0,0 +1,49 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class ReplaceCloakPalette : UpdateRule
+	{
+		public override string Name => "Change default Cloak style from Palette to Alpha.";
+
+		public override string Description =>
+			"Cloak has gained several new rendering modes\n" +
+			"and its default behaviour has changed from using a palette to native alpha.";
+
+		readonly List<string> locations = new();
+
+		public override IEnumerable<string> AfterUpdate(ModData modData)
+		{
+			if (locations.Count > 0)
+				yield return "Cloak no longer defaults to replacing the actor's palette.\n" +
+					"If you wish to keep the previous behavior you wish to change the\n" +
+					"Cloak definitions on the following actor definitions:\n" +
+					UpdateUtils.FormatMessageList(locations);
+
+			locations.Clear();
+		}
+
+		public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNodeBuilder actorNode)
+		{
+			foreach (var cloak in actorNode.ChildrenMatching("Cloak"))
+			{
+				cloak.RemoveNodes("Palette");
+				cloak.RemoveNodes("IsPlayerPalette");
+				locations.Add($"{actorNode.Key} ({actorNode.Location.Filename})");
+			}
+
+			yield break;
+		}
```

Comment:
```suggestion
				readonly Dictionary<string, string> actorsWithDefault = new();
		IEnumerable<string> IBeforeUpdateActors.BeforeUpdateActors(ModData modData, List<MiniYamlNodeBuilder> resolvedActors)
		{
			foreach (var actor in resolvedActors)
				foreach (var cloak in actor.ChildrenMatching("Cloak"))
					if (cloak.LastChildMatching("Palette", false) == null)
						actorsWithDefault.Add(actor.Key, cloak.Key);

			yield break;
		}

		public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNodeBuilder actorNode)
		{
			foreach (var cloak in actorNode.ChildrenMatching("Cloak"))
			{
				actorsWithDefault.TryGetValue(actorNode.Key, out var cloakKey);
				if (cloakKey == cloak.Key)
				{
					cloak.AddNode("CloakedPalette", "cloak");
					cloak.AddNode("CloakStyle", "Palette");
					yield break;
				}

				var palette = cloak.LastChildMatching("Palette", false);
				if (palette != null)
				{
					if (string.IsNullOrEmpty(palette.Value.Value))
					{
						cloak.RemoveNode(palette);
						cloak.AddNode("CloakStyle", "None");
					}
					else
					{
						palette.RenameKey("CloakedPalette");
						cloak.AddNode("CloakStyle", "Palette");
					}
				}
			}
		}
```

then something like this should do it

Reply:
IMO that is now getting needlessly complicated, especially considering that the desired changes for most mods (which I imagine simply copy RA's cloak) is to instead remove the `cloak` palette definition.

Reply:
That's for the mods to decide no? As for update rules they are meant to specifically preserve the old behaviour as much as possible

Reply:
People who use the resource centre updating for example don't care about manual changes, they just want to play